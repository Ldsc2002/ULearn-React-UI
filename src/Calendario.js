import React, { Component } from 'react';
import './App.css';
import ScreenContext from './ScreenContext';
import btn_icon_210788 from './images/btn_icon_210788.png';

// UI framework component imports
import Button from 'muicss/lib/react/button';

export default class Calendario extends Component {

  static contextType = ScreenContext;

  // This component doesn't use any properties

  constructor(props) {
    super(props);
    
    this.state = {
    };
  }

  componentDidMount() {
  }

  componentWillUnmount() {
  }

  componentDidUpdate(prevProps, prevState) {
  }

  render() {
    
    const style_elBackgroundShape = {
      background: 'rgba(255, 255, 255, 1.000)',
     };
    const style_elText = {
      color: 'black',
      textAlign: 'center',
     };
    
    const style_elFab = {
      display: 'block',
      textTransform: 'none',
      color: '(null)',
      textAlign: 'center',
     };
    
    return (
      <div className="Calendario">
        <div className="background">
          <div className="containerMinHeight elBackgroundShape" style={style_elBackgroundShape} />
        </div>
        
        <div className="layoutFlow">
          <div className="elText">
            <div className="baseFont" style={style_elText}>
              <div>{this.context.locStrings.calendario_text_722344}</div>
            </div>
          </div>
        </div>
        
        <div className="foreground">
          <Button className="actionFont elFab" style={style_elFab}  variant="fab" color="accent" >
            <img src={btn_icon_210788} alt="" style={{width: '50.000%', marginTop: '25.000%'}} />
          </Button>
        </div>
      </div>
    )
  }
  
}
