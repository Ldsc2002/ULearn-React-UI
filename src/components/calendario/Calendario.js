import React, { Component } from 'react'
import ScreenContext from '../app/ScreenContext'
import CalendarioComp from './CalendarioComp'

// UI framework component imports

export default class Calendario extends Component {
    // This component doesn't use any properties
    static contextType = ScreenContext;

    constructor(props) {
        super(props)

        this.state = {
        }
    }

    render() {
        return (
            <div className="Calendario">
                <div className="container has-text-centered">
                    <CalendarioComp />
                </div>
            </div>
        )
    }
}
