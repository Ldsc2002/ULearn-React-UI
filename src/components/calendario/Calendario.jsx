import React, { Component } from 'react'
import CalendarioComp from './CalendarioComp.jsx'
import ScreenContext from '../app/ScreenContext'

export default class calendario extends Component {
    static contextType = ScreenContext

    /*istanbul ignore next*/
    render() {
        return (
            <div className="Calendario">
                <div className="container has-text-centered">
                    <CalendarioComp
                        email={this.context.userInfo.email}
                    />
                </div>
            </div>
        )
    }
}
