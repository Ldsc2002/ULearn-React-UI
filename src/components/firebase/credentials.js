import { useState } from 'react'
import {
    createUserWithEmailAndPassword,
    signInWithEmailAndPassword,
    onAuthStateChanged,
    signOut,
} from 'firebase/auth'
import { auth } from './firebase'

const [registerEmail, setRegisterEmail] = useState('')
const [registerPassword, setRegisterPassword] = useState('')
const [loginEmail, setLoginEmail] = useState('')
const [loginPassword, setLoginPassword] = useState('')
const [user, setUser] = useState({})

onAuthStateChanged(auth, (currentUser) => {
    setUser(currentUser)
})

const register = async () => {
    try {
        await createUserWithEmailAndPassword(
            auth,
            registerEmail,
            registerPassword,
        )
    } catch (error) {
        alert(error)
    }
}

const login = async () => {
    try {
        await signInWithEmailAndPassword(
            auth,
            loginEmail,
            loginPassword,
        )
    } catch (error) {
        alert(error)
    }
}

const logout = async () => {
    await signOut(auth)
}
