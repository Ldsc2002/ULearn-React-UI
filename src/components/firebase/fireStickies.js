
import React, { useState, useEffect } from 'react';

function App() {
    const [input, setInput] = useState('');
    const [todos, setTodos] = useState([]);
  
    useEffect(() => { 
      db.collection('todos').onSnapshot(snapshot => {
        setTodos(snapshot.docs.map(doc => ({id:doc.id, todo:doc.data().todo})))
      })
    }, []);
  
    const createTodo = (e) => {
      e.preventDefault();
      db.collection('todos').add({
        todo: input
      });
      setInput('');
    };