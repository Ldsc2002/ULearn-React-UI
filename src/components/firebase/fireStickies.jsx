import { useState, useEffect } from 'react'
import { db } from './firebase'

function App() {
    const [input, setInput] = useState('')
    const [todos, setTodos] = useState([])

    useEffect(() => {
        db.collection('todos').onSnapshot((snapshot) => {
            setTodos(snapshot.docs.map((doc) => ({ id: doc.id, todo: doc.data().todo })))
        })
    }, [])

    const createTodo = (e) => {
        e.preventDefault()
        db.collection('todos').add({
            todo: input,
        })
        setInput('')
    }
}
