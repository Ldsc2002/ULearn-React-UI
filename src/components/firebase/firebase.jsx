import firebase from 'firebase/compat/app'
import 'firebase/compat/firestore'
import 'firebase/compat/auth'
import { getStorage } from 'firebase/storage'

const firebaseConfig = {
    apiKey: 'AIzaSyDtufOv6ScvHdl-SGog3CQmdEhtjl8oX6Q', 
    authDomain: 'ulearn-42d0d.firebaseapp.com',
    databaseURL: 'https://ulearn-42d0d-default-rtdb.firebaseio.com/',
    projectId: 'ulearn-42d0d',
    storageBucket: 'ulearn-42d0d.appspot.com', 
    messagingSenderId: '897319107558',
    appId: '1:1:897319107558:web:ba8d79e49bbd752e0ef427'
}

let app = ''

if (!firebase.apps.length) {
    app = firebase.initializeApp(firebaseConfig)
} else {
    app = firebase.app()
}

// const app = firebase.initializeApp(firebaseConfig);

const auth = firebase.auth(app)
const db = firebase.firestore()
const storage = getStorage()

export { auth, db, storage }
