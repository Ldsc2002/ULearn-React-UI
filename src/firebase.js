import firebase from 'firebase/compat/app'
import 'firebase/compat/firestore';
import 'firebase/compat/auth';
import 'dotenv/config'

const firebaseConfig = {
    apiKey: process.env.REACT_APP_API_KEY,
    authDomain: process.env.REACT_APP_AUTH_DOMAIN,
    databaseURL: process.env.REACT_APP_DATABASE_URL,
    projectId: process.env.REACT_APP_PROJECT_ID,
    storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
    messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
    appId: process.env.REACT_APP_APP_ID,
};

let app = ""

if (!firebase.apps.length) {
  app =firebase.initializeApp(firebaseConfig);
} else {
  app =firebase.app(); 
}

//const app = firebase.initializeApp(firebaseConfig);

const auth = firebase.auth(app);
const db = firebase.firestore();

export {auth,db}


