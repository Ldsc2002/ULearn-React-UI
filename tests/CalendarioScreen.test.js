import { render } from '@testing-library/react';
import React from 'react';
import Calendario from '../src/components/calendario/CalendarioComp';
import { db } from '../src/components/firebase/firebase';

jest.mock('firebase/auth', () => {
  return {
      createUsrWithEmailAndPassword: jest.fn(),
      updateProfile: jest.fn(),
      signInWithEmailAndPassword: jest.fn(),
  };
});

jest.mock('firebase/compat/app', () => {
  return {
      apps: [],
      firebase: jest.fn(),
      initializeApp: jest.fn(),
      auth: jest.fn(),
      firestore: jest.fn(),
      db: jest.fn(),
      storage: jest.fn(),
  };
});

jest.mock('firebase/compat/firestore', () => {
  return jest.fn();
});

jest.mock('firebase/compat/auth', () => {
  return jest.fn();
});

jest.mock('firebase/storage', () => {
  return{
      getStorage: jest.fn(),
  };
});


jest.mock('../src/components/firebase/firebase', () => {
  return {
      auth: jest.fn(),
      db: jest.fn(),
  };
});

//test to check if the calendario component renders correctly
test('calendario_render_test', () => {
  db.collection = jest.fn().mockReturnValue({
            get: jest.fn().mockReturnValue({
                then:jest.fn().mockResolvedValueOnce({})
            })
});
  render (<Calendario/>)

});

