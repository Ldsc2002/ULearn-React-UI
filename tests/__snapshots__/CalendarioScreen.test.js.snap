// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendario_render_test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="mountTitle"
      >
         Noviembre - 2022
      </p>
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              Lunes
            </th>
            <th>
              Martes
            </th>
            <th>
              Miércoles
            </th>
            <th>
              Jueves
            </th>
            <th>
              Viernes
            </th>
            <th>
              Sábado
            </th>
            <th>
              Domingo
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="in-prev-month "
            >
              31
            </td>
            <td
              class="in-this-month "
            >
              1
            </td>
            <td
              class="in-this-month "
            >
              2
            </td>
            <td
              class="in-this-month "
            >
              3
            </td>
            <td
              class="in-this-month "
            >
              4
            </td>
            <td
              class="in-this-month today"
            >
              5
            </td>
            <td
              class="in-this-month "
            >
              6
            </td>
          </tr>
          <tr>
            <td
              class="in-this-month "
            >
              7
            </td>
            <td
              class="in-this-month "
            >
              8
            </td>
            <td
              class="in-this-month "
            >
              9
            </td>
            <td
              class="in-this-month "
            >
              10
            </td>
            <td
              class="in-this-month "
            >
              11
            </td>
            <td
              class="in-this-month "
            >
              12
            </td>
            <td
              class="in-this-month "
            >
              13
            </td>
          </tr>
          <tr>
            <td
              class="in-this-month "
            >
              14
            </td>
            <td
              class="in-this-month "
            >
              15
            </td>
            <td
              class="in-this-month "
            >
              16
            </td>
            <td
              class="in-this-month "
            >
              17
            </td>
            <td
              class="in-this-month "
            >
              18
            </td>
            <td
              class="in-this-month "
            >
              19
            </td>
            <td
              class="in-this-month "
            >
              20
            </td>
          </tr>
          <tr>
            <td
              class="in-this-month "
            >
              21
            </td>
            <td
              class="in-this-month "
            >
              22
            </td>
            <td
              class="in-this-month "
            >
              23
            </td>
            <td
              class="in-this-month "
            >
              24
            </td>
            <td
              class="in-this-month "
            >
              25
            </td>
            <td
              class="in-this-month "
            >
              26
            </td>
            <td
              class="in-this-month "
            >
              27
            </td>
          </tr>
          <tr>
            <td
              class="in-this-month "
            >
              28
            </td>
            <td
              class="in-this-month "
            >
              29
            </td>
            <td
              class="in-this-month "
            >
              30
            </td>
            <td
              class="in-next-month "
            >
              1
            </td>
            <td
              class="in-next-month "
            >
              2
            </td>
            <td
              class="in-next-month "
            >
              3
            </td>
            <td
              class="in-next-month "
            >
              4
            </td>
          </tr>
          <tr>
            <td
              class="in-next-month "
            >
              5
            </td>
            <td
              class="in-next-month "
            >
              6
            </td>
            <td
              class="in-next-month "
            >
              7
            </td>
            <td
              class="in-next-month "
            >
              8
            </td>
            <td
              class="in-next-month "
            >
              9
            </td>
            <td
              class="in-next-month "
            >
              10
            </td>
            <td
              class="in-next-month "
            >
              11
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="megacontenedordebotones8000"
      >
        <button
          class="buttonControl"
          type="button"
        >
          Previo
        </button>
        <button
          class="buttonControl"
          type="button"
        >
          Siguiente
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      class="mountTitle"
    >
       Noviembre - 2022
    </p>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Lunes
          </th>
          <th>
            Martes
          </th>
          <th>
            Miércoles
          </th>
          <th>
            Jueves
          </th>
          <th>
            Viernes
          </th>
          <th>
            Sábado
          </th>
          <th>
            Domingo
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="in-prev-month "
          >
            31
          </td>
          <td
            class="in-this-month "
          >
            1
          </td>
          <td
            class="in-this-month "
          >
            2
          </td>
          <td
            class="in-this-month "
          >
            3
          </td>
          <td
            class="in-this-month "
          >
            4
          </td>
          <td
            class="in-this-month today"
          >
            5
          </td>
          <td
            class="in-this-month "
          >
            6
          </td>
        </tr>
        <tr>
          <td
            class="in-this-month "
          >
            7
          </td>
          <td
            class="in-this-month "
          >
            8
          </td>
          <td
            class="in-this-month "
          >
            9
          </td>
          <td
            class="in-this-month "
          >
            10
          </td>
          <td
            class="in-this-month "
          >
            11
          </td>
          <td
            class="in-this-month "
          >
            12
          </td>
          <td
            class="in-this-month "
          >
            13
          </td>
        </tr>
        <tr>
          <td
            class="in-this-month "
          >
            14
          </td>
          <td
            class="in-this-month "
          >
            15
          </td>
          <td
            class="in-this-month "
          >
            16
          </td>
          <td
            class="in-this-month "
          >
            17
          </td>
          <td
            class="in-this-month "
          >
            18
          </td>
          <td
            class="in-this-month "
          >
            19
          </td>
          <td
            class="in-this-month "
          >
            20
          </td>
        </tr>
        <tr>
          <td
            class="in-this-month "
          >
            21
          </td>
          <td
            class="in-this-month "
          >
            22
          </td>
          <td
            class="in-this-month "
          >
            23
          </td>
          <td
            class="in-this-month "
          >
            24
          </td>
          <td
            class="in-this-month "
          >
            25
          </td>
          <td
            class="in-this-month "
          >
            26
          </td>
          <td
            class="in-this-month "
          >
            27
          </td>
        </tr>
        <tr>
          <td
            class="in-this-month "
          >
            28
          </td>
          <td
            class="in-this-month "
          >
            29
          </td>
          <td
            class="in-this-month "
          >
            30
          </td>
          <td
            class="in-next-month "
          >
            1
          </td>
          <td
            class="in-next-month "
          >
            2
          </td>
          <td
            class="in-next-month "
          >
            3
          </td>
          <td
            class="in-next-month "
          >
            4
          </td>
        </tr>
        <tr>
          <td
            class="in-next-month "
          >
            5
          </td>
          <td
            class="in-next-month "
          >
            6
          </td>
          <td
            class="in-next-month "
          >
            7
          </td>
          <td
            class="in-next-month "
          >
            8
          </td>
          <td
            class="in-next-month "
          >
            9
          </td>
          <td
            class="in-next-month "
          >
            10
          </td>
          <td
            class="in-next-month "
          >
            11
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="megacontenedordebotones8000"
    >
      <button
        class="buttonControl"
        type="button"
      >
        Previo
      </button>
      <button
        class="buttonControl"
        type="button"
      >
        Siguiente
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
